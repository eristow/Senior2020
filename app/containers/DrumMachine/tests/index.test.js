/**
 *
 * Tests for DrumMachine
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { DrumMachine } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<DrumMachine />', () => {
  let store;
  const location = { store: {} };

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    // const Tone = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <DrumMachine dispatch={dispatch} location={location} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <DrumMachine location={location} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
