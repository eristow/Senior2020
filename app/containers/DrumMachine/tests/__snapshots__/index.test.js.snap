// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrumMachine /> Should render and match the snapshot 1`] = `
<div
  class="DrumMachine__Container-sc-2d7mbo-0 eIvoJm"
>
  <input
    class="InputText-wtwjzw-0 hWoTih"
    font-size="1.5em"
    type="text"
    value=""
    width="auto"
  />
  <div
    class="Transport__Wrapper-sc-16zg5lv-0 gCMBEA"
  >
    <div
      class="DrumMachine__ControlContainer-sc-2d7mbo-1 emVpvr"
    >
      <p
        class="DrumMachine__ControlText-sc-2d7mbo-2 hBLIRD"
      >
        BPM
      </p>
      <input
        class="BPMInput__BPM-sc-83ibx4-0 bTTBUp"
        max="180"
        min="50"
        type="number"
        value="80"
      />
    </div>
    <div
      class="DrumMachine__ControlContainer-sc-2d7mbo-1 emVpvr"
    >
      <p
        class="DrumMachine__ControlText-sc-2d7mbo-2 hBLIRD"
      >
        Kit
      </p>
      <select
        class="Dropdown-sc-1lo2juv-0 gDYBDL"
        width="5em"
      >
        <option
          value="config1"
        >
          EDM
        </option>
        <option
          value="config2"
        >
          Rock
        </option>
        <option
          value="config3"
        >
          Trap
        </option>
      </select>
    </div>
    <div
      class="DrumMachine__ControlContainer-sc-2d7mbo-1 emVpvr"
    >
      <p
        class="DrumMachine__ControlText-sc-2d7mbo-2 hBLIRD"
      >
        Master Volume
      </p>
      <div
        style="display: inline-block;"
      >
        <div
          class="rc-slider"
          style="width: 110px; height: 5px; margin: 0.5em 0.5em; z-index: 0;"
        >
          <div
            class="rc-slider-rail"
          />
          <div
            class="rc-slider-track"
            style="left: 0%; width: 100%;"
          />
          <div
            class="rc-slider-step"
          />
          <div
            aria-disabled="false"
            aria-valuemax="0"
            aria-valuemin="-60"
            aria-valuenow="0"
            class="rc-slider-handle"
            role="slider"
            style="left: 100%; transform: translateX(-50%);"
            tabindex="0"
          />
          <div
            class="rc-slider-mark"
          />
        </div>
      </div>
    </div>
    <div>
      <button
        class="PlayButton__Play-sc-1sbrdge-0 hCDzRT"
      >
        Play
      </button>
      <div>
        <button
          class="SaveButton__Save-sc-1uv51v8-0 iDlxTc"
        >
          Save
        </button>
      </div>
    </div>
  </div>
  <p>
    loading
  </p>
</div>
`;
